arbos/l1pricing/l1pricing.go

```diff
@@ -5,242 +5,358 @@ package l1pricing
+		sto.OpenStorageBackedUint64(lastUpdateTimeOffset),
+		sto.OpenStorageBackedBigInt(availableFundsOffset),
+		sto.OpenStorageBackedBigInt(fundsDueToSequencerOffset),
+		sto.OpenStorageBackedBigInt(fundsDueForRewards),
+		sto.OpenStorageBackedUint64(unitsSinceOffset),
+		sto.OpenStorageBackedBigInt(pricePerUnitOffset),
 	}
 }
 
-func (ps *L1PricingState) DefaultAggregator() (common.Address, error) {
-	return ps.defaultAggregator.Get()
+func (ps *L1PricingState) Sequencer() (common.Address, error) {
+	return ps.sequencer.Get()
 }
 
-func (ps *L1PricingState) SetDefaultAggregator(val common.Address) error {
-	return ps.defaultAggregator.Set(val)
+func (ps *L1PricingState) SetSequencer(seq common.Address) error {
+	return ps.sequencer.Set(seq)
 }
 
-func (ps *L1PricingState) L1BaseFeeEstimateWei() (*big.Int, error) {
-	return ps.l1BaseFeeEstimate.Get()
+func (ps *L1PricingState) PaySequencerFeesTo() (common.Address, error) {
+	return ps.paySequencerFeesTo.Get()
 }
 
-func (ps *L1PricingState) SetL1BaseFeeEstimateWei(val *big.Int) error {
-	return ps.l1BaseFeeEstimate.Set(val)
+func (ps *L1PricingState) SetPaySequencerFeesTo(addr common.Address) error {
+	return ps.paySequencerFeesTo.Set(addr)
 }
 
-func (ps *L1PricingState) LastL1BaseFeeUpdateTime() (uint64, error) {
-	return ps.lastL1BaseFeeUpdateTime.Get()
+func (ps *L1PricingState) PayRewardsTo() (common.Address, error) {
+	return ps.payRewardsTo.Get()
 }
 
-func (ps *L1PricingState) SetLastL1BaseFeeUpdateTime(t uint64) error {
-	return ps.lastL1BaseFeeUpdateTime.Set(t)
+func (ps *L1PricingState) EquilibrationTime() (uint64, error) {
+	return ps.equilibrationTime.Get()
 }
 
-// Update the pricing model with info from the start of a block
-func (ps *L1PricingState) UpdatePricingModel(baseFeeSample *big.Int, currentTime uint64) {
+func (ps *L1PricingState) Inertia() (uint64, error) {
+	return ps.inertia.Get()
+}
 
-	if baseFeeSample.Sign() == 0 {
-		// The sequencer's normal messages do not include the l1 basefee, so ignore them
-		return
-	}
+func (ps *L1PricingState) SetInertia(inertia uint64) error {
+	return ps.inertia.Set(inertia)
+}
 
-	// update the l1 basefee estimate, which is the weighted average of the past and present
-	//     basefee' = weighted average of the historical rate and the current, discounting time passed
-	//     basefee' = (memory * basefee + sqrt(passed) * sample) / (memory + sqrt(passed))
-	//
-	baseFee, _ := ps.L1BaseFeeEstimateWei()
-	inertia, _ := ps.L1BaseFeeEstimateInertia()
-	lastTime, _ := ps.LastL1BaseFeeUpdateTime()
-	if currentTime <= lastTime {
-		return
-	}
-	passedSqrt := arbmath.ApproxSquareRoot(currentTime - lastTime)
-	newBaseFee := arbmath.BigDivByUint(
-		arbmath.BigAdd(arbmath.BigMulByUint(baseFee, inertia), arbmath.BigMulByUint(baseFeeSample, passedSqrt)),
-		inertia+passedSqrt,
-	)
-
-	_ = ps.SetL1BaseFeeEstimateWei(newBaseFee)
-	_ = ps.SetLastL1BaseFeeUpdateTime(currentTime)
+func (ps *L1PricingState) PerUnitReward() (uint64, error) {
```

Comment:
There's no setter for this. We should probably add an `ArbOwner` method

Reply:
Done